cpp/ql/lib/semmle/code/cpp/exprs/Literal.qll

```diff
@@ -190,10 +190,23 @@ class ClassAggregateLiteral extends AggregateLiteral {
    * Gets the expression within the aggregate literal that is used to initialize
    * field `field`, if present.
    */
-  Expr getFieldExpr(Field field) {
+  Expr getFieldExpr(Field field) { result = this.getFieldExpr(field, _) }
+
+  /**
+   * Gets the expression within the aggregate literal that is used to initialize
+   * field `field` the `repitition`'th time in the initializer list, if present.
+   *
+   * For example, if `aggr` represents the initialization literal `{.x = 1234, .x = 5678}` in
+   * ```cpp
+   * struct Foo { int x; };
+   * struct Foo foo = {.x = 1234, .x = 5678};
+   * ```
+   * then `aggr.getFieldExpr(x, 0)` gives `1234`, and `aggr.getFieldExpr(x, 1)` gives `5678`.
+   */
+  Expr getFieldExpr(Field field, int repitition) {
     field = classType.getAField() and
     aggregate_field_init(underlyingElement(this), unresolveElement(result), unresolveElement(field),
-      _)
+      repitition)
```

Comment:
`repitition ` is not correct here. Suppose we have `{.x = 1234, .y = 5678}`, then we'll also get indexes `0` and `1` while there's no repetition (also not the correct spelling btw 😄 )

Reply:
:doh:

Reply:
Fixed in 52682e2ccbbf6c0965c48c41492deee9449d644b, I think. I have no idea why I started thinking this was the number of repetitions as I was working through the PR (maybe I was just tired!?). In any case, I've renamed all the repetition columns to position.

Reply:
I also changed the default `0` values to `-1` in IR generation as defaulting to the 0 position isn't a good choice.